gdal/frmts/grib/gribdataset.cpp

```diff
@@ -2287,15 +2309,34 @@ void GRIBDataset::SetGribMetaData(grib_MetaData *meta)
         if (rPixelSizeY < 0 || fabs(rPixelSizeY - meta->gds.Dy) > 0.002)
             rPixelSizeY = meta->gds.Dy;
 
-        // Longitude origin of GRIB files is sometimes funny. Try to shift as close
-        // as possible to the traditional [-180,180] longitude range
-        // See https://trac.osgeo.org/gdal/ticket/7103
-        if( ((rMinX >= 179 && rPixelSizeX * meta->gds.Nx > 10) || rMinX >= 180) &&
+        // GRIB2 files have longitudes in the [0-360] range
+        // Shift them to the traditional [-180,180] longitude range
+        // See https://github.com/OSGeo/gdal/issues/4524
+        if (meta->GribVersion == 2 &&
+            (rMinX + rPixelSizeX >= 180 || rMaxX >= 180) && 
```

Comment:
I would change the rMaxX >= 180 to a more relaxed one. Perhaps rMaxX - rPixelSizeX >= 180